### DISCUSSION TITLE
Text Generation Stops / Hangs and Stop Button Doesn't Work

### DISCUSSION THREAD

**Discussion by gibru** [open-webui/open-webui#2881] [General]
# Bug Report
Text Generation Hangs

**Bug Summary:**
Using Open WebUI, when prompting a model (e.g. Llama 70B), text generation stops after a while and the stop button is also unresponsive. Context length both tested default and 8K. Reloading the page allows to chat again, but the unfinished answer isn't saved. Using ollama directly doesn't have this issue (screenshots provided for comparison).

**Steps to Reproduce:**
Prompt the model for longer answers.

**Expected Behavior:**
The LLM is supposed to complete the answer and/or the stop button should allow for ending the text generation.

## Environment

- **Open WebUI Version:0.2.5**
- **Ollama (if applicable):1.4.1**
- **Operating System: Linux / Docker Installation**

## Logs and Screenshots

Screenshots attached

**Open WebUI example:**
![openwebui_prompt](https://github.com/open-webui/open-webui/assets/14093742/86a5ccf0-e6bc-48bb-b8b9-4793b14758d1)

**ollama (CLI) example:**
![ollama_prompt](https://github.com/open-webui/open-webui/assets/14093742/62fd791b-7045-4495-a3d3-2191687ae82b)

**Comment by TheTerrasque** [comment]
This is hopefully fixed with PR https://github.com/open-webui/open-webui/pull/3107

**Reply by gibru** [reply]
Thank you! And my apologies for not having included a console log (had to go read through the documentation first). Just in case, here's what my console log says the moment of the freeze:

`Uncaught (in promise) TypeError: Error in input stream
    $e Chat.svelte:367
    AsyncFunctionThrow self-hosted:803
    (Async: async)
    B MessageInput.svelte:721
`

**Comment by gibru** [comment]
Unfortunately, the error persists with the latest Open WebUI (`0.3.5`) and Ollama (`1.4.4`). Chat keeps freezing as reported above and the console log shows:

`Uncaught (in promise) TypeError: Error in input stream Ie Chat.svelte:376 AsyncFunctionThrow self-hosted:803`

I did some more testing besides comparing Ollama CLI vs Open WebUI. While Ollama CLI works flawlessly with all the models I'm testing (including the 70B ones), Open WebUI is only affected by the freeze when using 70B models like Llama 3 70B (q4 K_M) or qwen2:72b-instruct. While both of these models work fine with Ollama and Open WebUI's versions below 0.2.0, starting with 0.2.x+ they are are affected by the freeze. I do push it a bit with my measly 12GB of VRAM and 64GB of RAM, but that was never a problem before (besides being a little slow).

I guess long story short: where would I have to start looking to figure out what's going on? (I'm not a dev, but I'm curious enough to make an effort with a bit of guidance).

**Reply by TheTerrasque** [reply]
In my PR I set it to 15 minute timeout, but it was changed back to 5 minutes. If you look at the 0.3.5 release info, it says:

⏳ AIOHTTP_CLIENT_TIMEOUT: Introduced a new environment variable 'AIOHTTP_CLIENT_TIMEOUT' for requests to Ollama lasting longer than 5 minutes. Default is 300 seconds; set to blank ('') for no timeout.

You need to set that environment variable with longer (or no) timeout to avoid this problem.

**Reply by gibru** [reply]
I did make the make the necessary adjustment based on the [docs](https://docs.openwebui.com/getting-started/env-configuration) and changed the config.py inside the docker installation at `/app/backend` as follows: `AIOHTTP_CLIENT_TIMEOUT = os.environ.get("AIOHTTP_CLIENT_TIMEOUT", "")` . I'm happy to report that this has actually fixed the issue. Thanks a lot for your patience and help!

**Reply by TheTerrasque** [reply]
While you can change the config.py file inside docker, it's not really the recommended approach - and will probably reset if the container restarts. 

You can set env variables when starting a docker container, how it's done depends a bit on your setup and how you launch it. But https://www.cherryservers.com/blog/set-docker-environment-variables has some variations.

**Reply by gibru** [reply]
The learning never ceases! Thanks for the link. Another thing taken care of. Personal preferences and comfort lie more with system containers (lxc/incus) than Docker, but I guess I'll have to dig a bit deeper when it comes to Docker (or maybe even Podman).
